pages/book/security-best-practices.mdx

```diff
@@ -0,0 +1,144 @@
+# Security Best Practices
+
+In Tact smart contracts there are various anti-patterns and potential attacks developers need to be cautious of. These can impact the security, efficiency, and correctness of the contracts. Below, are covered both common anti-patterns and attacks specific to Tact smart contracts.
+
+## Send private data on chain
+
+Everything is stored on the blockchain. Remember that even if no private data is saved in the transaction, anyone can still get this data.
+
+## Incorrect use of signed/unsigned integer
+
+The following is an example of incorrect use of a signed integer. In the `Vote` message, the `votes` type is `int32`. This can lead to fraud when the attacker sends the negative number of votes instead of positive one.
+
+Invalid example:
+
+```tact
+message Vote {
+    votes: Int as int32;
+}
+
+contract Sample {
+    votes: Int as uint32 = 0;
+
+    receive(msg: Vote) {
+        self.votes += msg.votes;
+    }
+}
+```
+
+Unsigned integers are safer because they will error out if an overflow occurs, and signed integers are only used when they are really needed.
+
+
+## Invalid throw values
+
+Each time the TVM execution stops normally, it stops with exit codes `0` or `1`.
+Although it is done automatically, TVM execution can be interrupted directly in an unexpected way if exit codes `0` and `1` are thrown directly by either `throw(0)` or `throw(1)` command.
```

Comment:
```suggestion
## Throwing exit codes of normal execution

Exit codes $0$ and $1$ indicate normal TVM execution of the compute phase of the transaction. Execution can be unexpectedly aborted by calling [`throw(){:tact}`](https://docs.tact-lang.org/ref/core-debug#throw) (or [similar functions](https://docs.tact-lang.org/ref/core-debug)) directly with exit codes $0$ and $1$, which can make debugging very difficult since this aborted execution would be considered normal.
```